src/main/java/org/traccar/protocol/TranSyncProtocolDecoder.java

```diff
@@ -0,0 +1,218 @@
+package org.traccar.protocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.channel.Channel;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.Protocol;
+import org.traccar.helper.BitUtil;
+import org.traccar.helper.UnitsConverter;
+import org.traccar.model.CellTower;
+import org.traccar.model.Network;
+import org.traccar.model.Position;
+import org.traccar.session.DeviceSession;
+
+import java.net.SocketAddress;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+public class TranSyncProtocolDecoder extends BaseProtocolDecoder {
+
+    public TranSyncProtocolDecoder(Protocol protocol) {
+        super(protocol);
+    }
+
+    public Date getDatefromIntegerParameters(int year, int month, int day, int hours, int minutes, int seconds) {
+
+        String dataString = ""+year + "-" + month + "-" + day + " " + hours + ":" + minutes + ":" + seconds;
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yy-MM-dd HH:mm:ss");
+        simpleDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+        try {
+            return simpleDateFormat.parse(dataString);
+
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    protected void setDeviceAlert(Position position, int alert) {
```

Comment:
We usually have `decodeAlarm` method for this. Check other decoders and please use the same pattern and naming.